packages/components/src/utils/hooks.ts

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useCallback, useRef, useMemo } from "react";
+
+import generateId from "./useId";
+import canUseDom from "./canUseDom";
+
+/**
+ * Returns unique string for the `id` attribute.
+ *
+ * @param id Provided id
+ * @param name Optional prefix
+ */
+const useUniqueId = (id?: string, name?: string) =>
+  useMemo(() => id ?? generateId(name), [id, name]);
+
+interface StaticClass {
+  update?(config?: {}): void;
+  destroy?(): void;
+}
+
+interface StaticClassConstructable<T extends StaticClass> {
+  new (ref: HTMLElement, config: {}): T;
+}
+
+/**
+ * Hook that is managing "static" class in a reactive way
+ *
+ * @param StaticClass Class that encapsulates DOM mutating behaviour
+ * @param config Object holding message passed to static class
+ */
+
+const useStatic = <T extends StaticClass>(

```

Comment:
Nerozdelilme toto radsej do samostatneho suboru - vsetko co sa tyka useStatic a nie je to prepouzitelne?